src/main/java/mmosii/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,68 @@
+package mmosii.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mmosii.bookstore.dto.shoppingcart.CartItemDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemRequestDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemUpdateDto;
+import mmosii.bookstore.dto.shoppingcart.ShoppingCartDto;
+import mmosii.bookstore.service.ShoppingCartService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management",
+        description = "Endpoints for managing user`s shopping carts")
+@RestController
+@RequiredArgsConstructor
+@Validated
+@RequestMapping(value = "/api/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasAuthority('USER')")
+    @GetMapping
+    @Operation(summary = "Get shopping cart",
+            description = "Retrieve user's shopping cart")
+    public ShoppingCartDto getShoppingCart() {
+        return shoppingCartService.getShoppingCart();
+    }
+
+    @PreAuthorize("hasAuthority('USER')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
```

Comment:
this isn't a correct status

this status means you create some new resource every time this endpoint is invoked

but in fact - you don't

for simplicity I would remore response status here and let this endpoint return HTTP 200

Reply:
not fixed